### ISSUE TITLE
bashls error in nodejs >=18.1

### ISSUE THREAD

**Issue by storopoli** [williamboman/nvim-lsp-installer#707]
### Problem description

New version of nodejs broke bashls.
See https://github.com/bash-lsp/bash-language-server/issues/353.

Some people are downgrading nodejs to 17.9 versions:
> A temporary solution for me is to downgrade nodejs from 18.1.0 to 17.9.0 on Arch Linux...

I don't know how to put specs  in lua code to bound nodejs to <18.1.

### Neovim version (>= 0.7)

```bash
NVIM v0.7.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

### Operating system/version

Linux archdesktop1 5.17.7-hardened1-2-hardened #1 SMP PREEMPT Sun, 15 May 2022 05:50:41 +0000 x86_64 GNU/Linux

### I've recently downloaded the latest plugin version of both nvim-lsp-installer and nvim-lspconfig

- [X] Yes

### Affected language servers

bashls

### Actual behavior

This is my full error in `$HOME/.cache/nvim/lsp.log`:

```bash
[ERROR][2022-05-18 06:11:36] .../vim/lsp/rpc.lua:420    "rpc"   "bash-language-server"  "stderr"    "node:internal/deps/undici/undici:4813\n            throw new TypeError(\"Failed to parse URL from \" + input, { cause: err });\n                  ^\n\nTypeError: Failed to parse URL from /home/storopoli/.local/share/nvim/lsp_servers/bashls/node_modules/bash-language-server/node_modules/web-tree-sitter/tree-sitter.wasm\n    at new Request (node:internal/deps/undici/undici:4813:19)\n    at Agent.fetch2 (node:internal/deps/undici/undici:5539:29)\n    ... 4 lines matching cause stack trace ...\n    at new Promise (<anonymous>)\n    at Function.init (/home/storopoli/.local/share/nvim/lsp_servers/bashls/node_modules/bash-language-server/node_modules/web-tree-sitter/tree-sitter.js:1:332)\n    at Object.<anonymous> (/home/storopoli/.local/share/nvim/lsp_servers/bashls/node_modules/bash-language-server/out/parser.js:16:22)\n    at Generator.next (<anonymous>) {\n  [cause]: TypeError [ERR_INVALID_URL]: Invalid URL\n      at new NodeError (node:internal/errors:377:5)\n      at URL.onParseError (node:internal/url:563:9)\n      at new URL (node:internal/url:643:5)\n      at new Request (node:internal/deps/undici/undici:4811:25)\n      at Agent.fetch2 (node:internal/deps/undici/undici:5539:29)\n      at Object.fetch (node:internal/deps/undici/undici:6370:20)\n      at fetch (node:internal/bootstrap/pre_execution:196:25)\n      at /home/storopoli/.local/share/nvim/lsp_servers/bashls/node_modules/bash-language-server/node_modules/web-tree-sitter/tree-sitter.js:1:15192\n      at /home/storopoli/.local/share/nvim/lsp_servers/bashls/node_modules/bash-language-server/node_modules/web-tree-sitter/tree-sitter.js:1:15413\n      at new Promise (<anonymous>) {\n    input: '/home/storopoli/.local/share/nvim/lsp_servers/bashls/node_modules/bash-language-server/node_modules/web-tree-sitter/tree-sitter.wasm',\n    code: 'ERR_INVALID_URL'\n  }\n}\n\nNode.js v18.2.0\n"
```

### Expected behavior

No errors.

### Installation log

```shell
[INFO  Wed 18 May 2022 06:09:54 AM -03] ...ed/nvim-lsp-installer/lua/nvim-lsp-installer/ui/init.lua:751: Installation completed server_name="bashls", success=true
```

### Healthcheck

```shell
nvim-lsp-installer: require("nvim-lsp-installer.health").check()
========================================================================
## nvim-lsp-installer report
  - OK: neovim version >= 0.7.0
  - WARNING: **Go**: not available
  - WARNING: **cargo**: not available
  - WARNING: **Ruby**: not available
  - WARNING: **RubyGem**: not available
  - WARNING: **Composer**: not available
  - WARNING: **PHP**: not available
  - WARNING: **javac**: not available
  - OK: **sh**: `Ok`
  - OK: **tar**: `tar (GNU tar) 1.34`
  - OK: **gzip**: `gzip 1.12`
  - OK: **curl**: `curl 7.83.1 (x86_64-pc-linux-gnu) libcurl/7.83.1 OpenSSL/1.1.1o zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.47.0`
  - OK: **wget**: `GNU Wget 1.21.3 built on linux-gnu.`
  - WARNING: **pip3**: not available
  - OK: **python3**: `Python 3.10.4`
  - OK: **node**: `v18.2.0`
  - OK: **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
  - OK: **julia**: `julia version 1.7.2`
  - OK: **java**: `Ok`
  - OK: **npm**: `8.5.5`
  - OK: GitHub API rate limit. Used: 4. Remaining: 56. Limit: 60. Reset: Wed 18 May 2022 07:09:25 AM -03.
```

**Comment by williamboman** [comment]
Hello! Hmm I'm not sure what could be done in this plugin. The LSP client will currently use whatever node version that is in path (it also has nothing to do with lsp-installer). 

Configuring which node executable to use can technically already be done by modifying the `PATH` environment when setting up the server via lspconfig:

```lua
lspconfig.bashls.setup {
  cmd_env = {
    PATH = "<my_node_location>:" .. vim.env.PATH
}
```

**Comment by polyzen** [comment]
This was fixed in [bash-ls 3.0.1](https://github.com/bash-lsp/bash-language-server/blob/master/server/CHANGELOG.md#301).

**Comment by storopoli** [comment]
This is fixed in bash-ls 3.0.3.
